tests/lib/tools/log-analyzer

```diff
@@ -0,0 +1,511 @@
+#!/usr/bin/env python3
+
+"""
+This tool reads a spread log and creates a file with all the data
+The output file includes the more importan information extracted
+from the log to be analyzed
+"""
+
+import argparse
+import json
+import os
+import re
+import sys
+
+# Info types
+ERROR_TYPE = 'Error'
+DEBUG_TYPE = 'Debug'
+WARN_TYPE = 'WARNING:'
+
+# Results
+FAILED_TYPE = 'Failed'
+ABORTED_TYPE = 'Aborted'
+SUCCESSFUL_TYPE = 'Successful'
+
+# Printable names
+ALL = 'all'
+NONE = 'none'
+ACTION = 'action'
+OPERATION = 'operation'
+INFO = 'info'
+ERROR = 'error'
+FAILED = 'failed'
+ABORTED = 'aborted'
+SUCCESSFUL = 'successful'
+
+RESULT = 'result'
+START = 'Found'
+SPREAD_FILE = 'spread.yaml'
+
+EXEC_VERBS = ['Preparing', 'Executing', 'Restoring']
+INFO_TYPES = [ERROR_TYPE, DEBUG_TYPE, WARN_TYPE]
+OPERATIONS = [
+    'Rebooting', 'Discarding', 'Allocating', 'Waiting',
+    'Allocated', 'Connecting', 'Connected', 'Sending'
+    ]
+RESULTS = ['Successful', 'Aborted', 'Failed']
+
+class Action:
+    """
+    Action represents the main spread tasks actions
+    The actions can be: Preparing, Executing and Restoring
+    """
+
+    def __init__(self, verb, task, date, time, source_line):
+        self.type = ACTION
+        self.verb = verb
+        self.time = time
+        self.date = date
+        self.task = task
+        self.source_line = source_line
+
+    def __repr__(self):
+        return self.source_line
+
+    def __dict__(self):
+        return {
+            'type': 'action',
+            'date': self.date,
+            'time': self.time,
+            'verb': self.verb,
+            'task': self.task
+            }
+
+
+class Result:
+    """
+    Result represents the results for a spread run
+    The results can be: Successful, failed and aborted
+    """
+
+    def __init__(self, result_type, level, number, date, time, detail, source_line):
+        self.type = RESULT
+        self.result_type = result_type
+        self.level = level
+        self.number = number
+        self.time = time
+        self.date = date
+        self.detail = detail
+        self.source_line = source_line
+
+    def __repr__(self):
+        if self.detail:
+            return '{}{}'.format(self.source_line, str(self.detail))
+        return self.source_line
+
+    def __dict__(self):
```

Comment:
If you use something more modern you don't need to write this or the repr function, most likely.